pingcap/tidb
# privilege: add DDL and DML privilege check for system tables

### What problem does this PR solve? 

Add DDL and DML privilege check for system tables in performance_schema and metrics_schema.

### What is changed and how it works?

Prevent users from executing `ALTER`, `DROP`, `INDEX`, `INSERT`, `UPDATE`, `DELETE` statements on predefined tables.
Privileges of those tables which are defined in these schema by users themselves are kept untouched.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
 ```
mysql> delete from events_statements_summary_by_digest;
ERROR 8121 (HY000): privilege check fail
```

Code changes

 - Has exported function/method change

Side effects

 - Breaking backward compatibility

Related changes

N/A

Release note

 - Forbid users to execute DDL and update/delete/insert predefined tables in performance_schema and metrics_schema.
- Compatibility declaration:  Executing DDL and update/delete/insert predefined tables in performance_schema and metrics_schema are not allowed any longer.
